packages/app/src/app/pages/Dashboard/Content/routes/Settings/components/TeamInfo.tsx

```diff
@@ -0,0 +1,217 @@
+import React from 'react';
+import css from '@styled-system/css';
+import {
+  Button,
+  Element,
+  Stack,
+  Text,
+  Input,
+  Textarea,
+  IconButton,
+  Badge,
+} from '@codesandbox/components';
+import { TeamAvatar } from 'app/components/TeamAvatar';
+import { useWorkspaceAuthorization } from 'app/hooks/useWorkspaceAuthorization';
+import { useWorkspaceSubscription } from 'app/hooks/useWorkspaceSubscription';
+import { useActions, useAppState } from 'app/overmind';
+
+export const TeamInfo: React.FC = () => {
+  const [editing, setEditing] = React.useState(false);
+  const [loading, setLoading] = React.useState(false);
+  const [file, setFile] = React.useState<{ name: string; url: string } | null>(
+    null
+  );
+
+  const {
+    activeTeamInfo: team,
+    dashboard: { teams },
+  } = useAppState();
+  const actions = useActions();
+
+  const { isTeamAdmin } = useWorkspaceAuthorization();
+  const { isFree } = useWorkspaceSubscription();
+
+  const getFile = async avatar => {
+    const url = await new Promise((resolve, reject) => {
+      const reader = new FileReader();
+      reader.onload = e => {
+        resolve(e.target.result);
+      };
+      reader.onerror = reject;
+      reader.readAsDataURL(avatar);
+    });
+
+    const stringUrl = url as string;
+
+    setFile({
+      name: avatar.name,
+      url: stringUrl,
+    });
+  };
+
+  const handleTeamNameChange = event => {
+    const { value } = event.target;
+
+    // Get the input and remove any whitespace from both ends.
+    const trimmedName = value?.trim() ?? '';
+
+    // Validate if the name input is filled with whitespaces.
+    if (!trimmedName) {
+      event.target.setCustomValidity('Team name is required.');
+    } else if (teams.find(t => t.name === trimmedName)) {
+      event.target.setCustomValidity(
+        'Name already taken, please choose a new name.'
+      );
+    } else {
+      event.target.setCustomValidity('');
+    }
+  };
+
+  const onSubmit = async event => {
+    event.preventDefault();
+
+    const name = event.target.name.value?.trim();
+    const description = event.target.description.value?.trim();
+
+    if (!name) {
+      return;
+    }
+
+    setLoading(true);
+    // no try/catch because setTeamInfo dispatches
+    // a notification toast on error.
+    await actions.dashboard.setTeamInfo({
+      name,
+      description,
+      file,
+    });
+    setEditing(false);
```

Comment:
Found that after submitting and clicking the edit button again everything was still loading and I wasn't able to change the name again. Setting loading state to false should resolve this.

```suggestion
    setLoading(false);
    setEditing(false);
```

Reply:
@tristandubbeld thank you for catching this! 🙇🏽‍♀️